core/lib/zksync_core/src/api_server/execution_sandbox/mod.rs

```diff
@@ -227,6 +231,13 @@ impl BlockArgs {
     }
 
     /// Loads block information from DB.
+    ///
+    /// # Important!
+    ///
+    /// It is assumed that `block_id` is present in Postgres or can be present in the future
+    /// (i.e., it does not refer to a pruned block). If called for a pruned block, the returned value
+    /// (specifically, `l1_batch_timestamp_s`) will be nonsensical. Hence, the caller must take care
+    /// of pruned blocks separately.
```

Comment:
Maybe it makes sense to pass `BlockStartInfo` here and do the check internally?
I can imagine that _in theory_ we can forget to do the check (or accidentally remove it) during some quickfix to the API.
It'll be a bit more verbose, but I'd prefer us to return an error in such case rather than serve incorrect information.